apps/vaporgui/VaporWidgets.h

```diff
@@ -0,0 +1,112 @@
+#ifndef VAPORWIDGETS_H 
+#define VAPORWIDGETS_H 
+
+#include <QWidget>
+#include <QDoubleValidator>
+
+#include "Combo.h"
+#include "ui_VPushButton.h"
+#include "ui_VComboBox.h"
+#include "ui_VCheckBox.h"
+#include "ui_VPathSelector.h"
+
+namespace Ui {
+class QSliderEdit;
+}
+
+
+class VPushButton: public QWidget, public Ui_VPushButton
+{
+    Q_OBJECT
+
+public:
+    VPushButton(
+        QWidget* parent, 
+        std::string labelText = "Label",
+        std::string buttonText = "Button"
+    );
+
+    void SetLabelText(  const std::string text );
+    void SetLabelText(  const QString text );
+    void SetButtonText( const std::string text );
+    void SetButtonText( const QString text );
```

Comment:
Line 29 - 32 where a new string is passed in, a reference type is preferred. By using references, the application does not need to make a copy. For complex classes, i.e., strings, this copy could be expensive.

Reply:
Got it.  You're right.  std::string isn't a built-in so pass by const reference makes sense.  Thanks for the catch.